### PR TITLE
Add PostAction property

### LINKED ISSUE #3821: PostAction is not setable on DotNetSettings
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool, Cake Frosting, Cake runner for .NET Core

### Cake version

2.1

### Operating system

Windows

### Operating system architecture

64-Bit

### CI Server

_No response_

### What are you seeing?

It's not possible to set an `Action<IProcess>` on the DotNetSettings.
In my use case it's required to the the StandardOutput from the started dotnet tool process.

### What is expected?

A new option on DotNetSettings object

### Steps to Reproduce

See above

### Output log

_No response_

### PR DESCRIPTION
**Pull request by twenzel**
Fixes #3821 

This PR extends the ToolSettings to provide options to configure and handle the started process.

Example:
```csharp
var settings = new Cake.Common.Tools.DotNet.Format.DotNetFormatSettings()
{
    PostAction = (process) => 
    {
        process.WaitForExit();
        var x = process.GetStandardError();
    },
    SetupProcessSettings = (info) => info.RedirectStandardError = true,
    VerifyNoChanges = true
};

context.DotNetFormatWhitespace("./", settings);
```

### CODE REVIEW

**Comment by twenzel** [comment]
@augustoproiete This is something I totally forgot for the 2.1 release 😞

File: src/Cake.Common/Tools/DotNet/DotNetSettings.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using Cake.Core.IO;
```
Comment (nils-a) [inline src/Cake.Common/Tools/DotNet/DotNetSettings.cs:6]: Is this change needed?

File: src/Cake.Core/Tooling/ToolSettings.cs [hunk 2]
```diff
@@ -109,5 +109,15 @@ public class ToolSettings
...
+
+        /// <summary>
+        /// Gets or sets a delegate which is executed after the process was started.
+        /// </summary>
+        public Action<IProcess> PostAction { get; set; }
+
+        /// <summary>
+        /// Gets or sets a delegate to configure the process settings.
+        /// </summary>
+        public Action<ProcessSettings> SetupProcessSettings { get; set; }
```
Comment (nils-a) [inline src/Cake.Core/Tooling/ToolSettings.cs:121]: `SetupProcessSettings` is missing a test.

**Review by nils-a — CHANGES_REQUESTED** [review]
@twenzel looks good. I think there is one test missing, and one file had only usings added.

File: src/Cake.Common/Tools/DotNet/DotNetSettings.cs (hunk 1 above)
Comment (twenzel) [reply src/Cake.Common/Tools/DotNet/DotNetSettings.cs]: Nope. Reverted.

File: src/Cake.Core/Tooling/ToolSettings.cs (hunk 2 above)
Comment (twenzel) [reply src/Cake.Core/Tooling/ToolSettings.cs:121]: Right. Tests added to ToolTests.

**Review by nils-a — APPROVED** [review]
LGTM!

**Comment by nils-a** [comment]
@twenzel your changes have been merged, thanks for your contribution :+1:
